addons/project/controllers/portal.py

```diff
@@ -543,3 +545,43 @@ def portal_my_task(self, task_id, report_type=None, access_token=None, project_s
             request.session['my_tasks_history'] = task_sudo.ids
         values = self._task_get_page_view_values(task_sudo, access_token, **kw)
         return request.render("project.portal_my_task", values)
+
+    @http.route('/project_sharing/attachment/add_image', type='http', auth='user', methods=['POST'], website=True)
+    def add_image(self, name, data, res_id, access_token=None, **kwargs):
+        res_model = 'project.task'
+        try:
+            task_sudo = self._document_check_access(res_model, int(res_id), access_token=access_token)
+            if not task_sudo.with_user(request.env.uid).project_id._check_project_sharing_access():
+                return request.not_found()
+        except (AccessError, MissingError):
+            raise UserError(_("The document does not exist or you do not have the rights to access it."))
+
+        IrAttachment = request.env['ir.attachment']
+
+        # Avoid using sudo when not necessary: internal users can create attachments,
+        # as opposed to public and portal users.
+        if not request.env.user._is_internal():
+            IrAttachment = IrAttachment.sudo()
+
+        values = IrAttachment._check_contents({
```

Comment:
the other option is to call `create` directly (which will call `_check_contents` anyway) and then check the mimetype that was computed and raise if not in the white list

Reply:
I found out that `sudo(False)` si not working for this case be cause of this check : 
https://github.com/odoo/odoo/blob/f47344175b3be85bc3846a774304cef3af3a333f/odoo/addons/base/models/ir_attachment.py#L456

so to use `sudo` and not bypass some use-cases (like SVG), I had to check content before creating